static/app/views/issueDetails/eventNavigation.tsx

```diff
@@ -0,0 +1,214 @@
+  const location = useLocation();
+  const navigate = useNavigate();
+  const organization = useOrganization();
+
+  const hasPreviousEvent = defined(event.previousEventID);
+  const hasNextEvent = defined(event.nextEventID);
+
+  const baseEventsPath = `/organizations/${organization.slug}/issues/${group.id}/events/`;
+
+  return (
+    <div>
+      <EventNavigationWrapper>
+        <Tabs>
+          <TabList hideBorder hideSelection showPressed>
+            {Object.keys(EventNavLabels).map(label => {
+              return (
+                <TabList.Item
+                  to={{
+                    pathname: normalizeUrl(baseEventsPath + label + '/'),
+                    query: {...location.query, referrer: `${label}-event`},
+                  }}
+                  key={label}
+                >
+                  {EventNavLabels[label]}
+                </TabList.Item>
+              );
+            })}
+          </TabList>
+        </Tabs>
+        <NavigationWrapper>
+          <Navigation>
+            <LinkButton
+              title={'Previous Event'}
+              aria-label="Previous Event"
+              borderless
+              size="sm"
+              icon={<IconChevron direction="left" />}
+              disabled={!hasPreviousEvent}
+              to={{
+                pathname: `${baseEventsPath}${event.previousEventID}/`,
+                query: {...location.query, referrer: 'previous-event'},
+              }}
+            />
+            <LinkButton
+              title={'Next Event'}
+              aria-label="Next Event"
+              borderless
+              size="sm"
+              icon={<IconChevron direction="right" />}
+              disabled={!hasNextEvent}
+              to={{
+                pathname: `${baseEventsPath}${event.nextEventID}/`,
+                query: {...location.query, referrer: 'next-event'},
+              }}
+            />
+          </Navigation>
+          <StyledButton
+            onClick={() => {
+              const searchTermWithoutQuery = omit(location.query, 'query');
+              navigate({
+                pathname: normalizeUrl(
+                  `/organizations/${organization.slug}/issues/${group.id}/events/`
+                ),
+                query: searchTermWithoutQuery,
+              });
+            }}
+            borderless
+            size="sm"
+          >
+            {t('View All Events')}
+          </StyledButton>
+        </NavigationWrapper>
+      </EventNavigationWrapper>
+      <Divider />
+      <EventInfoJumpToWrapper>
+        <EventInfo>
+          <EventID>
+            {tct('Event [eventId]', {eventId: getShortEventId(event.id)})}
+          </EventID>
+          <TimeSince date={event.dateCreated ?? event.dateReceived} />
+        </EventInfo>
+        <JumpTo>
+          <div>{t('Jump to:')}</div>
+          <ButtonBar>
+            {jumpToSections.map(jump => {
+              if (!document.getElementById(jump.section.replace('#', ''))) {
```

Comment:
This line seems a bit suspect. It may be possible for react to be rendering before this element has been painted to the page, which would cause this to fail. Might be better to check the event contents instead?

Reply:
yeah - let me try to come up with a better way of doing this